modules/rostests/apitests/shlwapi/IsQSForward.cpp

```diff
@@ -0,0 +1,371 @@
+
+    cmds[0].cmdf = 0;
+
+    cCmds = 0;
+    pGUID = NULL;
+    for (cmdID = LOW_VALUE; cmdID <= HIGH_VALUE; ++cmdID)
+    {
+        cmds[0].cmdID = cmdID;
+
+        ret1 = IsQSForward(*pGUID, cCmds, cmds);
+        ret2 = IsQSForwardMockup(pGUID, cCmds, cmds);
+        ok(ret1 == ret2, "cmdID: %ld (%ld vs %ld)\n", cmdID, ret1, ret2);
+
+        ret1 = IsQSForward(*pGUID, cCmds, NULL);
+        ret2 = IsQSForwardMockup(pGUID, cCmds, NULL);
+        ok(ret1 == ret2, "cmdID: %ld (%ld vs %ld)\n", cmdID, ret1, ret2);
+    }
+
+    cCmds = 1;
+    for (cmdID = LOW_VALUE; cmdID <= HIGH_VALUE; ++cmdID)
+    {
+        cmds[0].cmdID = cmdID;
+
+        pGUID = NULL;
+        ret1 = IsQSForward(*pGUID, cCmds, cmds);
+        ret2 = IsQSForwardMockup(pGUID, cCmds, cmds);
+        ok(ret1 == ret2, "cmdID: %ld (%ld vs %ld)\n", cmdID, ret1, ret2);
+
+        ret1 = ret2 = 0xDEADFACE;
+        bExcept1 = bExcept2 = FALSE;
+
+        _SEH2_TRY
+        {
+            ret1 = IsQSForward(*pGUID, cCmds, NULL);
+        }
+        _SEH2_EXCEPT(EXCEPTION_EXECUTE_HANDLER)
+        {
+            bExcept1 = TRUE;
+        }
+        _SEH2_END;
+
+        _SEH2_TRY
+        {
+            ret2 = IsQSForwardMockup(pGUID, cCmds, NULL);
+        }
+        _SEH2_EXCEPT(EXCEPTION_EXECUTE_HANDLER)
+        {
+            bExcept2 = TRUE;
+        }
+        _SEH2_END;
+
+        ok(ret1 == ret2, "cmdID: %ld (%ld vs %ld)\n", cmdID, ret1, ret2);
+        ok((bExcept1 && bExcept2), "cmdID: %ld (%d vs %d)\n", cmdID, bExcept1, bExcept2);
+
+        pGUID = &CGID_Explorer;
+        ret1 = IsQSForward(*pGUID, cCmds, cmds);
+        ret2 = IsQSForwardMockup(pGUID, cCmds, cmds);
+        ok(ret1 == ret2, "cmdID: %ld (%ld vs %ld)\n", cmdID, ret1, ret2);
+
+        pGUID = &IID_IUnknown;
+        ret1 = IsQSForward(*pGUID, cCmds, cmds);
+        ret2 = IsQSForwardMockup(pGUID, cCmds, cmds);
+        ok(ret1 == ret2, "cmdID: %ld (%ld vs %ld)\n", cmdID, ret1, ret2);
+    }
+
+    cCmds = 2;
+    for (cmdID = LOW_VALUE; cmdID <= HIGH_VALUE; ++cmdID)
+    {
+        cmds[0].cmdID = cmdID;
+
+        pGUID = NULL;
+        for (cmdID2 = LOW_VALUE; cmdID2 <= HIGH_VALUE; ++cmdID2)
+        {
+            cmds[1].cmdID = cmdID2;
+            ret1 = IsQSForward(*pGUID, cCmds, cmds);
+            ret2 = IsQSForwardMockup(pGUID, cCmds, cmds);
+            ok(ret1 == ret2, "cmdID: %ld (%ld vs %ld)\n", cmdID, ret1, ret2);
+        }
+
+        pGUID = &CGID_Explorer;
+        for (cmdID2 = LOW_VALUE; cmdID2 <= HIGH_VALUE; ++cmdID2)
+        {
+            cmds[1].cmdID = cmdID2;
+            ret1 = IsQSForward(*pGUID, cCmds, cmds);
+            ret2 = IsQSForwardMockup(pGUID, cCmds, cmds);
+            ok(ret1 == ret2, "cmdID: %ld (%ld vs %ld)\n", cmdID, ret1, ret2);
+        }
+    }
+}
+
+static VOID TEST_MayQSForwardMockup(VOID)
+{
+    CDummyClass dummy;
+    LONG cmdID;
+    OLECMD cmds[1];
+    OLECMDTEXT cmdText;
+    HRESULT ret1, ret2;
+
+    // OLECMDID_PROPERTIES
```

Comment:
Maybe: `// Testing OLECMDID_PROPERTIES support` ? (If this was the intent of your comment, then the other ones below may be adapted similarly.)

Reply:
Fixed them.